Mathlib/RingTheory/Ideal/Span.lean

```diff
@@ -257,3 +257,46 @@ theorem ker_toSpanSingleton_one_sub_eq_span :
   rw [ker_toSpanSingleton_eq_span he.one_sub, sub_sub_cancel]
 
 end IsIdempotentElem
+
+namespace Subideal
+
+open Subtype
+
+variable {A : Type*} [CommSemiring A] {I : Ideal A}
+
+/-- The Galois coinsertion between the map sending a subideal `J ≤ I` to itself,
+  and the mal sending `J ↦ J ⊓ I`. -/
+def galoisCoinsertion :
```

Comment:
@alreadydone, do we have this for Set.Iic?

Reply:
I can't find it in mathlib but here it is:
```
import Mathlib.Data.Set.Basic
import Mathlib.Order.GaloisConnection.Defs
import Mathlib.Order.Interval.Set.Defs
import Mathlib.Order.Lattice

variable {α : Type*} [SemilatticeInf α] (a : α)

example : GaloisCoinsertion ((↑) : Set.Iic a → α) fun a' : α ↦ ⟨a ⊓ a', inf_le_left⟩ where
  choice a' le := ⟨a', le.trans inf_le_left⟩
  gc b c := by
    simp_rw [← Subtype.coe_le_coe, le_inf_iff]; exact (and_iff_right b.2).symm
  u_l_le _ := inf_le_right
  choice_eq _ le := le_antisymm le inf_le_right
```

Reply:
I have deleted this since it is not used by the `CompleteLattice` instance anymore (and it is not needed either for our application in the divided powers project). Should I add the `Set.Iic` version in this PR? If so, do you know what would be the right file to add it to?